sway-book/src/documentation/language/functions/return.md

```diff
@@ -0,0 +1,39 @@
+# Returning from functions
+
+In the previous sections we have seen how functions return values without much information. In this section we will take a closer look at how we can return data from a function.
+
+There are two ways to return:
+
+<!-- no toc-->
+- [Explicitly](#explicit-return)
+- [Implicitly](#implicit-return)
+
+When returning data from a function the return types must match up with the return types declared in the function signature. This means that if the first return type is a `u64` then the type of the first value being returned must also be a `u64`.
+
+## Explicit Return
+
+To return from a function explicitly we use the `return` keyword followed by the arguments and a semi-colon.
+
+```sway
+{{#include ../../../code/language/functions/src/explicit.sw:main}}
+```
+
+A return expression is typically used at the end of a function however as long as the syntax is correct it can be used anywhere inside a function. This can be used as a mechanism to exit the function early if some condition is met.
+
+```sway
+{{#include ../../../code/language/functions/src/explicit.sw:return_data}}
+```
+
+## Implicit Return
+
+To return from a function implicitly we do not use the `return` keyword and we omit the semi-colon at the end of the line.
+
+```sway
+{{#include ../../../code/language/functions/src/implicit.sw:main}}
+```
+
+Similarly to the explicit usage of a `return` this will typically be used at the end of a function but it can also be used anywhere.
```

Comment:
A return expression will only return from the function if it is itself _inside an expression_ . If you're inside a statement (something ending with a `;`, more or less) then you need to explicitly `return;`

In my mental model, an "implicit return" is just a special case of the way expressions work in Sway (and Rust), rather than a thing in its own right.